testing.md

```diff
@@ -0,0 +1,93 @@
+# Testing
+
+## Test Runner (nextest)
+
+The default test runner works fine, but it is notably slower and less featureful than [nextest].
+
+Fortunately, [nextest] ships with recent versions of the compile-env, so assuming you have already followed the instructions in the [README.md](./README.md), you should be able to run
+
+```shell
+just cargo nextest run
+```
+
+even if you have not installed [nextest].
+
+:warning: [nextest profiles] are not the same thing as [cargo profiles].
+If you want to select a cargo profile when running [nextest], use, for example
+
+```shell
+just cargo nextest run --cargo-profile=release
+```
+
+## Code Coverage (llvm-cov)
+
+The compile-env also ships with [cargo llvm-cov] for collecting [code coverage](https://en.wikipedia.org/wiki/Code_coverage) information.
+Assuming you have followed the [README.md](./README.md), you should be able to run
+
+```shell
+just coverage
+```
+
+to get code coverage information.
+
+Code coverage reports from CI are uploaded to [our codecov page](https://app.codecov.io/gh/githedgehog/dataplane).
+
+If you wish to study coverage data locally, you can run 
+
+```shell
+just coverage
+cd ./target/nextest/coverage/html
+python3 -m http.server
+```
+
+And then open a web-browser to [http://localhost:8000](http://localhost:8000) to view coverage data.
+
+## Fuzz testing (bolero)
+
+The dataplane project makes fairly extensive use of [fuzz testing](https://en.wikipedia.org/wiki/Fuzzing).
+We use the [bolero] crate for our fuzz tests.
+
+Running the test suite via `just cargo test` or `just cargo nextest run` will run the fuzz tests, but it does so under significant limitations.
+
+1. The tests (even the fuzz tests) are only run briefly.
+2. Coverage information and sanitizers are not enabled.
+3. A full fuzzing engine is not set up, so evolutionary feedback is not provided when the tests are run this way,
```

Comment:
Note: I'd use unordered lists when there's no particular reason to order the items.

Reply:
Ok.